projects/step-core/src/lib/modules/routing/injectables/navigator.service.ts

```diff
@@ -86,7 +86,12 @@ export class NavigatorService {
     const queryParams = this.prepareQueryParams();
     const queryParamsHandling: QueryParamsHandling = !queryParams ? 'preserve' : '';
     const segments = link.split('/');
-    this._router.navigate(segments, { queryParams, queryParamsHandling });
+    const hasFilters = segments.some((el) => el.includes('?'));
```

Comment:
Invocation of `prepareQueryParams` few lines above, doesn't make any sense if the link contain query parameters.

1. Please move the following part to the else block:

```
const queryParams = this.prepareQueryParams();
const queryParamsHandling: QueryParamsHandling = !queryParams ? 'preserve' : '';
```

2. This method doesn't know anything about filters.
Better name for `hasFilters` would be something like `hasInlineQueryParameters`

3. What is the reason to check `?` character in each url segment? Isn't it's easier to do `link.includes('?')`?

Reply:
Code changed. Also moved 'const segments = link.split('/');' to else block as it's not needed if condition is true.